File: src/ProjectR/content/scripts/spells/Inspire.lua
function GetTargetType()
    return Single
end

function GetName()
    return "Inspire"
end

function GetDescription()
    return "Inspire target, slightly rasing their offense. Also restores some of casters MP."
end

function IsSupportSpell()
    return true
end

function GetMPCost()
    return 0
end

function GetDelay()
  return .5
end

function GetSpellType()
  return Physical
end

function GetMasteries()
  return {  }
end

function SpellEffect()
    d:BuffStat(AD, .2)
    d:BuffStat(MD, .2)
    a:UseMP(-(aMP * .2))
end

File: src/ProjectR.Scripting/content/scripts/spells/Inspire.cs
using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting.Spells
{
    public class Inspire : SpellScriptBase
    {
        public override string Name { get { return "Inspire"; } }
        public override string Description { get { return "Inspire target, slightly raising their offense. Also restores some of casters MP."; } }

        public override TargetType TargetType { get { return TargetType.Single; } }
        public override IList<EleMastery> Masteries { get { return new EleMastery[] {  }; } }
        public override SpellType SpellType { get { return SpellType.Physical; } }
        public override bool IsSupportSpell { get { return true; } }

        public override double MPCost { get { return 0; } }
        public override double Delay { get { return .5; } }

        public override void SpellEffect(ICharacter caster, ICharacter target)
        {
            BuffStat(Stat.AD, .2);
            BuffStat(Stat.MD, .2);
            caster.UseMP(-(cMP * .2));
        }
    }
}